validation-test/stdlib/Range.swift.gyb

```diff
@@ -410,21 +540,183 @@ ${TestSuite}.test("count/whereBoundIsStrideable") {
+    var idx = range.startIndex
+    var previousValue : Int? = nil
+
+    for _ in 0${op}(endValue - 1) {
+      // Advance the index to the last in-range position
+      idx = range.index(after: idx)
+      if let previousValue = previousValue {
+        expectEqual(range[idx].value, previousValue + 1)
+      }
+      previousValue = range[idx].value
+    }
+    // Iterate once more to get to the 'after end' position
+    idx = range.index(after: idx)
+    expectEqual(idx, range.endIndex)
+    // Now, iterate once more and expect a crash
+    expectCrashLater()
+    _ = range.index(after: idx)
+  }
+}
+
+${TestSuite}.test("index(after:)/semantics/smallestRange") {
+  let range: ${Self}<${Bound}> = ${Bound}(0)${op}${Bound}(0)
+  var idx = range.startIndex
+
+%     if 'Closed' in Self:
+  // Should be able to advance the range once
+  idx = range.index(after: idx)
+%     end
+
+  expectCrashLater()
+    _ = range.index(after: idx)
+}
+
+${TestSuite}.test("index(before:)/semantics") {
+  let rangeSizes = [3, 5, 10, 12, 86]
+
+  for endValue in rangeSizes {
+    let start = ${Bound}(0)
+    let end = ${Bound}(endValue)
+    let range: ${Self}<${Bound}> = start${op}end
+    var idx = range.endIndex
+    var previousValue : Int? = nil
+
+    for _ in 0${op}(endValue) {
+      // Advance the index backwards until we reach `startIndex`
+      // Precondition: for a..<b, advance endIndex (b - a) times to reach startIndex
+      // Precondition: for a...b, advance endIndex (b - a) + 1 times to reach startIndex
+      idx = range.index(before: idx)
+      if let previousValue = previousValue {
+        expectEqual(range[idx].value, previousValue - 1)
+      }
+      previousValue = range[idx].value
+    }
+    // At this point, we should be at `startIndex`
+    expectEqual(idx, range.startIndex)
+    // Now, iterate once more and expect a crash
+    expectCrashLater()
+    _ = range.index(before: idx)
+  }
+}
+
+${TestSuite}.test("index(before:)/semantics/smallestRange") {
+  let range: ${Self}<${Bound}> = ${Bound}(0)${op}${Bound}(0)
+  var idx = range.endIndex
+
+%     if 'Closed' in Self:
+  // Should be able to advance the range once
+  idx = range.index(before: idx)
+%     end
+
+  expectCrashLater()
+  _ = range.index(before: idx)
+}
+
+${TestSuite}.test("index(offsetBy:)/semantics") {
+  for test in advanceIndexTests {
+    let start = ${Bound}(AdvanceIndexTest.start)
+    let end = ${Bound}(AdvanceIndexTest.end)
+    let range: ${Self}<${Bound}> = start${op}end
+    let initialIdx = test.advanceFromEnd ? range.endIndex : range.startIndex
+
+%     if 'Closed' in Self:
+    if let newIndex = test.newClosedIndex {
+%     else:
+    if let newIndex = test.newOpenIndex {
+%     end
+      let idx = range.index(initialIdx, offsetBy: test.advanceBy)
+      if idx != range.endIndex {
+        expectEqual(newIndex, range[idx].value, stackTrace: SourceLocStack().with(test.loc))
+      }
+    } else {
+      expectCrashLater()
```

Comment:
This would crash the test and not execute the rest of the tests.  Please use parameterized tests.